### PR TITLE
feat: generate Flannel CNI manifest from upstream

### LINKED ISSUE #6730: update Flannel manifest from upstream
Use e.g. https://github.com/flannel-io/flannel/blob/v0.20.2/Documentation/kube-flannel.yml, and transform it to make it compatible with Talos.

We can have `go generate` step which takes upstream manifest, transforms and generates `.go` file with new manifest.

We should use "our" mirrors of upstream Flannel image and our `install-cni` init container.

Flannel manifest tag should match Flannel version used.

### PR DESCRIPTION
**Pull request by smira**
Fixes #6730

`go generate`-based step downloads the upstream manifest, transforms it to match our requirements, and it is compiled in as the Flannel manifest.

### CODE REVIEW

**Review by frezbo — APPROVED** [review]
:cool I was thinking o something similar for cilium, except generate in hack so helm deps dpn't get into talos machinery

**Review by frezbo — APPROVED** [review]
:cool I was thinking o something similar for cilium, except generate in hack so helm deps dpn't get into talos machinery

**Comment by smira** [comment]
```diff
 > processing manifest rbac.authorization.k8s.io/v1.ClusterRole/flannel
--- a/rbac.authorization.k8s.io/v1.ClusterRole/flannel
+++ b/rbac.authorization.k8s.io/v1.ClusterRole/flannel
@@ -14,6 +14,7 @@
   resources:
   - nodes
   verbs:
+  - get
   - list
   - watch
 - apiGroups:

 < applied successfully
 > processing manifest rbac.authorization.k8s.io/v1.ClusterRoleBinding/flannel
 < no changes
 > processing manifest v1.ServiceAccount/kube-system/flannel
 < no changes
 > processing manifest v1.ConfigMap/kube-system/kube-flannel-cfg
 < no changes
 > processing manifest apps/v1.DaemonSet/kube-system/kube-flannel
--- a/apps/v1.DaemonSet/kube-system/kube-flannel
+++ b/apps/v1.DaemonSet/kube-system/kube-flannel
@@ -2,7 +2,7 @@
 kind: DaemonSet
 metadata:
   annotations:
-    deprecated.daemonset.template.generation: "1"
+    deprecated.daemonset.template.generation: "2"
   labels:
     k8s-app: flannel
     tier: node
@@ -21,6 +21,15 @@
         k8s-app: flannel
         tier: node
     spec:
+      affinity:
+        nodeAffinity:
+          requiredDuringSchedulingIgnoredDuringExecution:
+            nodeSelectorTerms:
+            - matchExpressions:
+              - key: kubernetes.io/os
+                operator: In
+                values:
+                - linux
       containers:
       - args:
         - --ip-masq
@@ -38,21 +47,24 @@
             fieldRef:
               apiVersion: v1
               fieldPath: metadata.namespace
-        - name: POD_IP
-          valueFrom:
-            fieldRef:
-              apiVersion: v1
-              fieldPath: status.podIP
+        - name: EVENT_QUEUE_DEPTH
+          value: "5000"
         image: ghcr.io/siderolabs/flannel:v0.20.2
         imagePullPolicy: IfNotPresent
         name: kube-flannel
-        resources: {}
+        resources:
+          limits:
+            cpu: 100m
+            memory: 50Mi
+          requests:
+            cpu: 100m
+            memory: 50Mi
         securityContext:
           capabilities:
             add:
             - NET_ADMIN
             - NET_RAW
-          privileged: true
+          privileged: false
         terminationMessagePath: /dev/termination-log
         terminationMessagePolicy: File
         volumeMounts:
@@ -90,7 +94,8 @@
         terminationMessagePolicy: File
         volumeMounts:
         - mountPath: /host/opt/cni/bin/
-          name: host-cni-bin
+          name: cni-plugin
+      priorityClassName: system-node-critical
       restartPolicy: Always
       schedulerName: default-scheduler
       securityContext: {}
@@ -108,6 +113,10 @@
           type: ""
         name: run
       - hostPath:
+          path: /opt/cni/bin
+          type: ""
+        name: cni-plugin
+      - hostPath:
           path: /etc/cni/net.d
           type: ""
         name: cni
@@ -115,10 +124,6 @@
           defaultMode: 420
           name: kube-flannel-cfg
         name: flannel-cfg
-      - hostPath:
-          path: /opt/cni/bin
-          type: ""
-        name: host-cni-bin
   updateStrategy:
     rollingUpdate:
```

File: pkg/machinery/constants/constants.go [hunk 1]
```diff
@@ -817,6 +817,9 @@ const (
 
 	// DashboardTTY is the number of the TTY device (/dev/ttyN) for dashboard.
 	DashboardTTY = 2
+
```
Comment (frezbo) [inline pkg/machinery/constants/constants.go:820]: missed the part to add renovate comment

**Pull request merged**